packages/gatsby/content/advanced/migration.md

```diff
@@ -37,9 +37,26 @@ Run `npx @yarnpkg/doctor .` (or `yarn dlx @yarnpkg/doctor .`) in your project to
 
 Note that the doctor is intended to report any potential issue - it's then up to you to decide whether they are a false positive or not (for example it won't traverse Git repositories). For this reason we don't recommend using it as a CI tool.
 
-### Use `yarn dlx` instead of `yarn global`
-
-`yarn dlx` is designed to execute one off scripts that may have been installed as global packages with `yarn 1.x`. Managing system-wide packages is outside of the scope of `yarn`. To reflect this, `yarn global` has been removed. [Read more on GitHub](https://github.com/yarnpkg/berry/issues/821).
+### Changes in CLI commands
+
+| v1              | v2 substitute                                | Notes                                                               |
+| --------------- | -------------------------------------------- | ------------------------------------------------------------------- |
+| `yarn audit`    | -                                            | [Read more on GitHub](https://github.com/yarnpkg/berry/issues/1187) |
+| `yarn check`    | `yarn install --immutable --immutable-cache` | [Read more on GitHub](https://github.com/yarnpkg/rfcs/pull/106)     |
```

Comment:
`yarn check` never really did anything useful - the actual v1 command that `--immutable` mimics was `yarn install --frozen-lockfile`, although it has slightly different semantics (and the cache immutability is new).

Reply:
If I'm not mistaken, from the [yarn-check](https://classic.yarnpkg.com/en/docs/cli/check):

> ### # `yarn check --integrity`
> Verifies that versions and hashed values of the package contents in the project’s `package.json` match those in yarn’s lock file.

This corresponds to `--immutable`:

> If the `--immutable` option is set, Yarn will abort with an error exit code if anything in the install artifacts (`yarn.lock`, `.pnp.js`, ...) was to be modified.

And,

> ### # `yarn check --verify-tree`
> Recursively verifies that the dependencies in `package.json` are present in `node_modules` and have the right version. This check does not consider `yarn.lock`.

roughly corresponds to:

> If the `--immutable-cache` option is set, Yarn will abort with an error exit code if the cache folder was to be modified (either because files would be added, or because they'd be removed).

I think it's the closest to what's being documented.

Reply:
No; `--integrity` used to do what `yarn install` already does (we check the cache integrity at every install), and `--verify-tree` isn't needed because Yarn 2 never has wrong tree (or it's a bug, and users aren't expected to have to code defensively against this).

The `--immutable` flag is pretty much `--frozen-lockfile`, and `--immutable-cache` is kinda like `--offline`.

Reply:
Ok, I have emptied the substitute field.